subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultMavenArtifactRepository.java

```diff
@@ -239,6 +241,20 @@ public void metadataSources(Action<? super MetadataSources> configureAction) {
         configureAction.execute(metadataSources);
     }
 
+    @Override
+    public void setKind(RepositoryKind kind) {
+        Action<? super ArtifactResolutionDetails> filter = null;
+        switch (kind) {
+            case RELEASES_ONLY:
+                filter = ReleasesOnly.INSTANCE;
+                break;
+            case SNAPSHOTS_ONLY:
+                filter = SnapshotsOnly.INSTANCE;
+                break;
```

Comment:
Why not add a default case and set it to `RELEASES_ONLY`("RELEASES").

Reply:
Because it's a breaking change if you do so. Currently Gradle assumes Maven repositories are `MIXED`.